File: Backend/src/main/java/com/luckyseven/backend/sharedkernel/jwt/repository/RefreshTokenRepository.java
package com.luckyseven.backend.sharedkernel.jwt.repository;

import com.luckyseven.backend.sharedkernel.jwt.entity.RefreshToken;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RefreshTokenRepository extends JpaRepository<RefreshToken,Long> {


  Optional<RefreshToken> findByUserId(Long id);


  void deleteByTokenValue(String refreshToken);
}

File: Backend/src/main/kotlin/com/luckyseven/backend/sharedkernel/jwt/repository/RefreshTokenRepository.kt
package com.luckyseven.backend.sharedkernel.jwt.repository

import com.luckyseven.backend.sharedkernel.jwt.entity.RefreshToken
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.Instant

@Repository
interface RefreshTokenRepository : JpaRepository<RefreshToken, Long> {
    fun findByUserId(id: Long): RefreshToken?
    
    fun deleteByTokenValue(refreshToken: String)

    fun findAllByExpiresAtBefore(now: Instant): List<RefreshToken>
} 